lab3/client/src/views/generator.js

```diff
@@ -0,0 +1,77 @@
+import Generator from '../models/generator';
+
+export default function (store) {
+	return class GeneratorComponent extends window.HTMLElement {
+		constructor () {
+			super();
+			this.store = store;
+
+			// TODO: render generator initial view
+	
+			// TODO: subscribe to store on change event
+			this.onStateChange = this.handleStateChange.bind(this);
+
+			// TODO: add click event
+
+		}
+
+		handleStateChange(newState)
+		{
+			//this.innerHTML = this.init()
+			const gen = new Generator(newState.generators[this.dataset.id]);
+
+			this.innerHTML = `<h3><b>${gen.name}</b></h3>
+			<p>${gen.description}</p>
+			<button id="but">5 Skulls ${gen.generate}</button>
```

Comment:
You don't click on `button` to generate

The generator is supposed to generate automatically from homework 1.